New hs_docker_base image, new hs_postgres image

mjstealey opened the pull request:
Complete refactoring of the HydroShare base image and the PostgreSQL image
Please review:
1.  [hs_docker_base:new.base](https://github.com/hydroshare/hs_docker_base/tree/1.6.9)
2. [hs_postgres:9.4.7](https://github.com/hydroshare/hs_postgres/tree/master)

Update notes:
- Derived from the Python:2.7.11 official docker image
- Django 1.8.12
- Debian 8 (jessie) based
- PostgreSQL 9.4.7 (new hs_postgres repository built on docker hub)
    - PostgreSQL 9.4.7 on x86_64-unknown-linux-gnu, compiled by gcc (Debian 4.9.2-10) 4.9.2, 64-bit
    - POSTGIS="2.2.2 r14797" GEOS="3.4.2-CAPI-1.8.2 r3921" PROJ="Rel. 4.8.0, 6 March 2012" GDAL="GDAL 1.10.1, released 2013/08/26" LIBXML="2.9.1" LIBJSON="0.11.99" RASTER
- hstore
- All packages installed at their current release level unless absolutely unsupported by existing hydroshare code

hydrobuild commented:
Can one of the admins verify this patch?

selimnairb commented:
@mjstealey Can you bump Django to 1.8.12?

mjstealey commented:
> Can you bump Django to 1.8.12

Updated in the Dockerfile, rebuilding image currently in docker hub (takes about 35-40 mins).

mjstealey commented:
@selimnairb, @hyi, @pkdash, @ironfroggy - Would like to integrate this sooner than later.

- Please do review the new Dockerfiles for the base and postgres
- Slowly removing the fuse code (in favor of iRODS NFS which seems safer to me)
- Might still be lacking a few packages, but won't fully know until it's in the hands of the developers

pkdash commented on hsctl:
```diff
@@ -227,37 +229,42 @@ rebuild_nodb_hs() {
+    echo "*** INFO: CREATE EXTENSION postgis on PostgreSQL ***"
+    docker exec hydroshare psql -U postgres -h postgis -w -c 'CREATE EXTENSION postgis;'
+    echo "*** INFO: CREATE EXTENSION hstore on PostgreSQL ***"
+    docker exec hydroshare psql -U postgres -h postgis -w -c 'CREATE EXTENSION hstore;'
```
During my testing of hstore extension I had to run the following command in order for the test database to also have access to the hstore extension (note the template option):
   
     psql -U postgres -h postgis -d template1 -w -c 'create extension hstore;'

pkdash commented:
HStore also requires psycopg2 version >=2.5. The following line then needs change:
https://github.com/hydroshare/hs_docker_base/blob/develop/requirements.txt#L75

mjstealey commented:
> HStore also requires psycopg2 version >=2.5

It is already at **psycopg2==2.6.1**, please refer to the linked branch as denoted in the first comment (it is not the master branch, but rather [branch 1.6.9](https://github.com/hydroshare/hs_docker_base/tree/1.6.9))

selimnairb commented:
@mjstealey Is this running on beta yet?  If so, and if the tests pass, and the site is functioning properly, we should release to production.

mjstealey commented:
> Is this running on beta yet?

It's running on alpha at the moment

hyi commented:
@mjstealey I don't see any issues and agree with @selimnairb - if things are running properly and tests all pass, we should merge and deploy it.

mjstealey commented:
In the process of deploying to beta, and noticed the same error on alpha as well, migrations fail when encountering: **hs_geo_raster_resource.custom_migration_for_tif_to_vrt_20160223**

This causes the remaining migrations to not be run.

```bash
Running migrations:
  Rendering model states... DONE
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying blog.0001_initial... FAKED
  Applying blog.0002_auto_20150527_1555... OK
  Applying conf.0001_initial... FAKED
  Applying core.0001_initial... FAKED
  Applying core.0002_auto_20150414_2140... OK
  Applying django_comments.0001_initial... FAKED
  Applying django_comments.0002_update_user_email_field_length... OK
  Applying pages.0001_initial... FAKED
  Applying forms.0001_initial... FAKED
  Applying forms.0002_auto_20141227_0224... OK
  Applying forms.0003_emailfield... OK
  Applying forms.0004_auto_20150517_0510... OK
  Applying forms.0005_auto_20151026_1600... OK
  Applying galleries.0001_initial... FAKED
  Applying galleries.0002_auto_20141227_0224... OK
  Applying generic.0001_initial... FAKED
  Applying generic.0002_auto_20141227_0224... OK
  Applying hs_core.0017_auto_20160219_2039... OK
  Applying hs_core.0017_auto_20160217_1629... OK
  Applying hs_core.0018_merge... OK
  Applying hs_core.0019_baseresource_locked_time... OK
  Applying hs_core.0020_baseresource_collections... OK
  Applying hs_collection_resource.0001_initial... OK
  Applying hs_geo_raster_resource.custom_migration_for_tif_to_vrt_20160223...Deleted bag for resource ID:e660640a7b084794aa2d70dc77cfa67b
Deleted bag for resource ID:c813e5c26ba24222ba1b73f43943aca6
Deleted bag for resource ID:e438bf9871ec402b8f2cf27f37f10081
Deleted bag for resource ID:8defcfa0edd848daaf0ef25271fc6703
Deleted bag for resource ID:27c329b95d9646e593b75bd10ab71132
Deleted bag for resource ID:c9616269b5094c51b71632e8d1d02c0d
Deleted bag for resource ID:8d874553dc80475db60548586b30c9b8
Deleted bag for resource ID:318480bd6db94393ac59347802df9057
Deleted bag for resource ID:983fd49c63d04ac091388490f8bdd689
Deleted bag for resource ID:9a19cb02bef5470a8ed650934eca40ea
Deleted bag for resource ID:02f993e9170848fb82f44edaaa9a5b10
Deleted bag for resource ID:d23f6fb3e6b04c758c665964d1f73897
Deleted bag for resource ID:2f512b89f27a4f93b56935e9f460898e
Deleted bag for resource ID:d9141a6ae8dd4fd2866ab8d09da6b9ae
Deleted bag for resource ID:b64f56a1f7bf4385b20d5d2a476d6f8b
Deleted bag for resource ID:dc25e924c3f440b1b88c5d3f708f3253
Deleted bag for resource ID:eab608a147f446a4b4fd038800f841c5
Deleted bag for resource ID:639a897cd39749389bef0c952e37326a
Deleted bag for resource ID:a94af692e2d1442cad3797d23ed7cdf8
Deleted bag for resource ID:2f3aaaed72ff46ccad4814c7b8e20996
Deleted bag for resource ID:a9b005d9a0e345f0bd2fbcb0fd7195d5
Deleted bag for resource ID:650229a8d9904dd58053a29ba1ddbd46
Traceback (most recent call last):
  File "manage.py", line 8, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 222, in handle
    executor.migrate(targets, plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 110, in migrate
    self.apply_migration(states[migration], migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 148, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/migration.py", line 115, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/operations/special.py", line 183, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/home/docker/hydroshare/hs_geo_raster_resource/migrations/custom_migration_for_tif_to_vrt_20160223.py", line 25, in migrate_tif_file
    vrt_file_path, temp_dir = create_vrt_file(res_file.resource_file)
  File "/home/docker/hydroshare/hs_geo_raster_resource/receivers.py", line 90, in create_vrt_file
    subprocess.Popen(['gdalbuildvrt', vrt_file_path, tif_file.file.name], stdout=fp, stderr=fp).wait()   # remember to add .wait()
  File "/usr/local/lib/python2.7/site-packages/django/db/models/fields/files.py", line 51, in _get_file
    self._file = self.storage.open(self.name, 'rb')
  File "/usr/local/lib/python2.7/site-packages/django/core/files/storage.py", line 37, in open
    return self._open(name, mode)
  File "/home/docker/hydroshare/django_irods/storage.py", line 148, in _open
    self.session.run("iget", None, '-f', name, tmp.name)
  File "/home/docker/hydroshare/django_irods/icommands.py", line 171, in run
    raise SessionException(proc.returncode, stdout, stderr)
django_irods.icommands.SessionException: (SessionException(...), 'Error processing IRODS request: 3. stderr follows:\n\nERROR: resolveRodsTarget: srcPath /hydrotest41Zone/home/betaHydroProxy/15b0de6022c845548e993a1261250539/data/contents/cubdem.tif does not exist\nERROR: getUtil: resolveRodsTarget status = -317000 USER_INPUT_PATH_ERR\n')
*** manage.py fix_permissions ***
```

If the **hs_geo_raster_resource.custom_migration_for_tif_to_vrt_20160223** migrations file is deleted, all other migrations will complete and the site runs.

```bash
Running migrations:
  Rendering model states... DONE
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying blog.0001_initial... FAKED
  Applying blog.0002_auto_20150527_1555... OK
  Applying conf.0001_initial... FAKED
  Applying core.0001_initial... FAKED
  Applying core.0002_auto_20150414_2140... OK
  Applying django_comments.0001_initial... FAKED
  Applying django_comments.0002_update_user_email_field_length... OK
  Applying pages.0001_initial... FAKED
  Applying forms.0001_initial... FAKED
  Applying forms.0002_auto_20141227_0224... OK
  Applying forms.0003_emailfield... OK
  Applying forms.0004_auto_20150517_0510... OK
  Applying forms.0005_auto_20151026_1600... OK
  Applying galleries.0001_initial... FAKED
  Applying galleries.0002_auto_20141227_0224... OK
  Applying generic.0001_initial... FAKED
  Applying generic.0002_auto_20141227_0224... OK
  Applying hs_core.0017_auto_20160219_2039... OK
  Applying hs_core.0017_auto_20160217_1629... OK
  Applying hs_core.0018_merge... OK
  Applying hs_core.0019_baseresource_locked_time... OK
  Applying hs_core.0020_baseresource_collections... OK
  Applying hs_collection_resource.0001_initial... OK
  Applying pages.0002_auto_20141227_0224... OK
  Applying pages.0003_auto_20150527_1555... OK
  Applying theme.0002_auto_20160219_2039... OK
  Applying theme.0003_auto_20160302_0453... OK
*** manage.py fix_permissions ***
```

I believe this may be due to the **irsync** issue when copying the www resources to other servers as noted last week.

I also suspect this wouldn't be an issue on www as all resources should be found, and it may not even be necessary as I believe this only needed to run once, though it would be nice if the migration could continue if a specific file is not found instead of crashing.
